openproject/client.py

```diff
@@ -0,0 +1,96 @@
+import httpx
+from exceptions import APIError, AuthenticationError
+from openproject.types import WorkPackage
+
+
+class Client:
+    def __init__(self, base_url: str, api_token: str):
+        self.base_url = base_url
+        self.api_version = "v3"
+        self.api_token = api_token
+
+    def _handle_response(self, response: httpx.Response):
+        if response.status_code == 401:
+            raise AuthenticationError(response.json()["message"], response.status_code)
```

Comment:
Let's not assume that `message` exists and name the variable `error` as we do below:

```suggestion
            error = response.json().get("message", "Authentication error")
            raise AuthenticationError(error, response.status_code)
```